fix: Only look at the current branch files' status when looking for concurrent modifications

carols10cents opened the pull request:
And, notably, ignore any new Parquet files that now appear in the catalog for this partition.

Also, don't treat this as an error yet-- only log that this happens, so we can check the logic without affecting production behavior.

NGA-TRAN commented on compactor2/src/partition_info.rs:
```diff
@@ -45,27 +45,36 @@ impl PartitionInfo {
+
+        // If there are any files in `self`/`old` that are not present in `new`, that means some files were marked
+        // to delete by some other process.
+        missing.next().is_some()
```
If the level is different, it will be in this missing, right?

carols10cents replied:
Yes, as shown in the test `changed_compaction_level_indicates_modification` in the commit I just added-- could you take a look and let me know if that's clear?

NGA-TRAN reviewed (approved):
Makes sense. I looked at the tests, too

carols10cents commented:
Ok, I removed the commit that changed the error filtering and added a commit that removes the error entirely, making a new component, and there's only a logging implementer of the component right now (so it's a little weird because there's no `inner` component the logging component is wrapping).

I also made the log message only print the IDs and compaction levels of the files that were in the `branch` that don't appear in the catalog set when we went to commit; this should make the log message smaller than the skipped_compaction reason that contained all of both old and new 😅 

Please review this carefully to make sure this code handles everything!!

NGA-TRAN commented on compactor2/src/components/changed_files_filter/mod.rs:
```diff
@@ -0,0 +1,213 @@
+        let pf_id3_level_1 = ParquetFileBuilder::new(3)
+            .with_compaction_level(CompactionLevel::FileNonOverlapped)
+            .build();
+        let pf_id4_level_0 = ParquetFileBuilder::new(1)
```
Do you mean id = 4?

carols10cents replied:
Great catch, I will fix that!

NGA-TRAN commented:
It looks good to me. 
@Joe-Blount : you many want to have a look to see if you want it to only go to log for now as you suggested

Joe-Blount commented:
> you many want to have a look to see if you want it to only go to log for now as you suggested

that's how it is with the current commits :+1:

carols10cents commented:
Ok, let's try this again... 

![Jeff Goldblum in Independence Day saying "without the oops"](https://user-images.githubusercontent.com/193874/228956233-12e5c652-982f-4282-9361-ca1302d8f9aa.gif)
